services/blob/service/src/main/java/org/collectionspace/services/blob/BlobResource.java

```diff
@@ -149,18 +150,54 @@ public InputStream getBlobContent(ServiceContext<PoxPayloadIn, PoxPayloadOut> ct
     	} catch (Exception e) {
     		throw bigReThrow(e, ServiceMessages.CREATE_FAILED);
     	}
-    	
+
     	if (result == null) {
     		String errMsg = String.format("Index failed. Could not get the contents for the Blob with CSID = '%s'.",
     				csid);
 	        Response response = Response.status(
 	                Response.Status.INTERNAL_SERVER_ERROR).entity(errMsg).type("text/plain").build();
 	        throw new CSWebApplicationException(response);
     	}
-    	
+
     	return result;
     }
-    
+
+    private File getBlobContentFile(ServiceContext<PoxPayloadIn, PoxPayloadOut> ctx,
+                                    String csid,
+                                    String derivativeTerm,
+                                    StringBuffer outMimeType) throws CSWebApplicationException {
+        File result = null;
+
+        try {
+            BlobInput blobInput = BlobUtil.getBlobInput(ctx);
+            blobInput.setDerivativeTerm(derivativeTerm);
+            blobInput.setContentRequested(true);
+
+            PoxPayloadOut response = this.get(csid, ctx);
+            logger.debug(response.toString());
+
+            // The result of a successful get should have put the results in the blobInput instance
+            String mimeType = blobInput.getMimeType();
+            if (mimeType != null) {
+                outMimeType.append(mimeType); // blobInput's mime type was set on call to "get" above by the doc handler
+            }
+            result = BlobUtil.getBlobInput(ctx).getBlobFile();
```

Comment:
@ray-lee After doing some testing I found some issues with the blob api from these changes. My first try at testing this was with a plain text file, and when trying to retrieve it I was returned an HTTP 500. I can still see that the Blob has a content stream and I can see the file on disk, so I'm not sure what's going on that the file isn't being set properly. Testing against dev I can retrieve a plain text blob.

Other types of media do work as expected (e.g. jpegs), but we'll want to get this fixed before merging.

Reply:
Nice find. Thanks for the thorough testing.

Reply:
I made an update to change `File getBlobContentFile` to `ResponseBuilder getBlobContentResponse`. If a Blob doesn't have a File it will fall back to the InputStream, which won't be able to have the Range header applied and will return the full content stream.

I can make a ticket to track adding partial content support to plain/text. When looking into this yesterday I didn't see a quick way as we're relying on Nuxeo's Blob which doesn't provide a File for certain types.

Reply:
@mikejritter @ray-lee When CollectionSpace hands Nuxeo a text/plain file, Nuxeo creates a "Note" document which is backed by Nuxeo's StringBlob class.  The StringBlob class always returns null as the value to the getFileBlob() method.  

If we need/want to support partial content requests for text/plain files, we'll need to change how CollectionSpace asks Nuxeo to create the corresponding blob.

From UCB perspective, there's **no** need to support partial content requests for text/plain files, so Mike's fix should be fine.